### DISCUSSION TITLE
SWR useRequest function using Axios with hard to use `mutate` function

### DISCUSSION THREAD

**Discussion by hoetmaaiers** [vercel/swr#695] [Q&A]
When trying to combine SWR with Axios, I like the approach I found in the examples: https://github.com/vercel/swr/blob/master/examples/axios-typescript/libs/useRequest.ts

This way my axios config resembles the keyInterface, this is great.
The problem starts when I try to mutate my data. In the bound mutate function, the expected data includes the whole AxiosResponse object, which makes no sense when locally mutating.

Am I doing something wrong with the demo? Or am I misusing  the mutate functionality?

### Demo:
I use the basic `useRequest` hook from the examples (link above). When I act on the mutate function returned from `useRequest()` like this: `mutate({ data: updatedAnswers}, false);`, I receive this compiler error:

```
Argument of type '{ data: Answer[]; }' is not assignable to parameter of type 'AxiosResponse<Answer[]> | Promise<AxiosResponse<Answer[]>> | mutateCallback<AxiosResponse<Answer[]>> | undefined'.
  Type '{ data: Answer[]; }' is missing the following properties from type 'AxiosResponse<Answer[]>': status, statusText, headers, config
```

**Comment by csauvage** [comment]
I'm running into the same issue.
Maybe @shyamlohar @MoeCasts or @Svish can help ? 
Thanks.

**Reply by Svish** [reply]
We have currently only used `swr` for data fetching, so no experience with `mutate` at all. Sorry.

**Comment by hoetmaaiers** [comment]
Sorry, I replace SWR in favor of React-Query.
